lib/musl.rb

```diff
@@ -0,0 +1,98 @@
+end
+MUSL_ENV_OPTIONS = "PATH=#{CREW_MUSL_PREFIX}/bin:#{ENV['PATH']} \
+    CC='#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-gcc' \
+    CXX='#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-g++' \
+    LD='#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-ld.gold -L#{CREW_MUSL_PREFIX}/lib' \
+    AR=#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-ar \
+    LIBRARY_PATH=#{CREW_MUSL_PREFIX}/lib \
+    MUSL_PKG_CONFIG='#{MUSL_PKG_CONFIG}' \
+    PKG_CONFIG_LIBDIR=#{CREW_MUSL_PREFIX}/lib/pkgconfig \
+    CFLAGS='#{MUSL_CFLAGS}' \
+    CXXFLAGS='#{MUSL_CXXFLAGS}' \
+    CPPFLAGS='-I#{CREW_MUSL_PREFIX}/include -fcommon -idirafter#{CREW_PREFIX}/include' \
+    LDFLAGS='#{MUSL_LDFLAGS}'"
+MUSL_CMAKE_OPTIONS = "PATH=#{CREW_MUSL_PREFIX}/bin:#{CREW_MUSL_PREFIX}/#{ARCH}-linux-musl#{MUSL_ABI}/bin:#{ENV['PATH']} \
+      CC='#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-gcc' \
+      CXX='#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-g++' \
+      LD=#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-ld.gold \
+      AR=#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-ar \
+      MUSL_PKG_CONFIG='#{MUSL_PKG_CONFIG}' \
+      PKG_CONFIG_LIBDIR=#{CREW_MUSL_PREFIX}/lib/pkgconfig \
+      CFLAGS='#{MUSL_CFLAGS}' \
+      CXXFLAGS='#{MUSL_CXXFLAGS}' \
+      CPPFLAGS='-I#{CREW_MUSL_PREFIX}/include -fcommon -idirafter#{CREW_PREFIX}/include' \
+      LDFLAGS='#{MUSL_CMAKE_LDFLAGS}' \
+      cmake \
+      -DCMAKE_INSTALL_PREFIX='#{CREW_MUSL_PREFIX}' \
+      -DCMAKE_INSTALL_LIBDIR='#{CREW_MUSL_PREFIX}/lib' \
+      -DCMAKE_LIBRARY_PATH='#{CREW_MUSL_PREFIX}/lib' \
+      -DCMAKE_C_COMPILER=#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-gcc \
+      -DCMAKE_CXX_COMPILER=#{CREW_MUSL_PREFIX}/bin/#{ARCH}-linux-musl#{MUSL_ABI}-g++ \
+      -DCMAKE_INCLUDE_DIRECTORIES_BEFORE=ON \
+      -DINCLUDE_DIRECTORIES=#{CREW_MUSL_PREFIX}/include \
+      -DCMAKE_C_FLAGS='#{MUSL_CFLAGS}' \
+      -DCMAKE_CXX_FLAGS='#{MUSL_CXXFLAGS}' \
+      -DCMAKE_EXE_LINKER_FLAGS='#{MUSL_CMAKE_LDFLAGS}' \
+      -DCMAKE_SHARED_LINKER_FLAGS='#{MUSL_CMAKE_LDFLAGS}' \
+      -DCMAKE_STATIC_LINKER_FLAGS='#{MUSL_CMAKE_LDFLAGS}' \
+      -DCMAKE_MODULE_LINKER_FLAGS='#{MUSL_CMAKE_LDFLAGS}' \
+      -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=TRUE \
+      -DCMAKE_BUILD_TYPE=Release"
+
+$VERBOSE = warn_level
+
+class Musl
+  class << self
+    def patchelf
+      puts 'Running Musl.patchelf'.lightblue
+      abort('No Patchelf found!').lightred unless File.exist?("#{CREW_PREFIX}/bin/patchelf")
+
+      @execfiles = `find . -executable -type f ! \\( -name \"*.a\" \\) -exec sh -c \"file -i \'{}\' | grep -q \'executable; charset=binary\'\" \\; -exec ls -1i {} \\; | sort -u -n -s -k1,1 | awk '{print $2}'`.chomp
+      return if @execfiles.empty? || @execfiles.nil?
+
+      puts 'Running patchelf to patch binaries for musl paths'.lightblue
+      @execfiles.each_line do |execfiletopatch|
+        execfiletopatch.slice! '.'
```

Comment:
Will `.delete_prefix` better (since it will not break files that contains `.` in the filename)

Reply:
Let me try that!

Reply:
`execfiletopatch = Dir.pwd + execfiletopatch.delete_prefix('.')` looks like it does the trick!